src/java/htsjdk/samtools/MergingSamRecordIterator.java

```diff
@@ -144,17 +145,6 @@ public SAMRecord next() {
             }
         }
 
-        // Fix up the sequence indexes if needs be
-        if (this.samHeaderMerger.hasMergedSequenceDictionary()) {
-            if (record.getReferenceIndex() != SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
-                record.setReferenceIndex(this.samHeaderMerger.getMergedSequenceIndex(iterator.getReader().getFileHeader(), record.getReferenceIndex()));
-            }
-
-            if (record.getReadPairedFlag() && record.getMateReferenceIndex() != SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
-                record.setMateReferenceIndex(this.samHeaderMerger.getMergedSequenceIndex(iterator.getReader().getFileHeader(), record.getMateReferenceIndex()));
-            }
-        }
```

Comment:
I don' think we can leave this code in with the current changes - it assumes the reference indices are stale and relative to the old dictionary, and tries to resolve them by translation. With these changes, the setHeader call above will resolve the indices (by name) against the new dictionary, so they'll already be translated. Treating them as if they're in the old space would be incorrect I think. I should be able to add a test to verify this.